contracts/Staking.sol

```diff
@@ -0,0 +1,45 @@
+pragma solidity 0.6.2;
+
+import "./vendor/SafeMath.sol";
+import "./StorageManager.sol";
+
+contract Staking {
+
+    using SafeMath for uint256;
+
+    StorageManager storageManager;
+
+    mapping(address => uint256) internal amountStaked;
+
+    event Staked(address indexed user, uint256 amount, uint256 total, bytes data);
+    event Unstaked(address indexed user, uint256 amount, uint256 total, bytes data);
+
+    constructor(address _storageManager) public {
```

Comment:
here we can do:
`constructor(StorageManager _storageManager) public {`
and then simply assign. Not sure which is the better approach but I usually see this pattern being used.

Reply:
We can do that. The reason I prefer using address is because of simplicity. (Rhetorical question: do you know how to pass an ERC20 to a constructor in Remix, or with Javascript?).

I don't know, and I suspect more people don't. Which is why I prefer to always just pass an address.

Reply:
But I'm open to be proven wrong here in this choice ;)

Reply:
Its OK.. its just a type safety measure, you simply pass an `address` from outside no need to do anything different but we can leave it as-is.